scripts/o3de/o3de/download.py

```diff
@@ -124,9 +124,11 @@ def download_o3de_object(object_name: str, default_folder_name: str, dest_path:
     if not dest_path:
         logger.error(f'Destination path cannot be empty.')
         return 1
-    if dest_path.exists():
+    if dest_path.exists() and not force_overwrite:
         logger.error(f'Destination path {dest_path} already exists.')
         return 1
+    elif dest_path.exists():
+        shutil.rmtree(dest_path)
```

Comment:
one fun gotcha with rmtree is that if it fails it throws an exception, and another is that when users use Perforce it is common for folders that are checked in to be set to read-only, so rmtree() will fail here too, but you will find [workarounds for this using the onerror param](https://stackoverflow.com/questions/2656322/shutil-rmtree-fails-on-windows-with-access-is-denied).  I think we should at least catch exception and give the user a friendly log error message.  You should be able to test this by setting the gem folder to read only in Windows explorer.